src/base.ts

```diff
@@ -0,0 +1,31 @@
+import { UpdateTypeInterface } from './types/updateTypeInterface';
+
+export abstract class TypeBase implements UpdateTypeInterface {
+  eq(str1: string, str2: string): boolean {
+    return this.compare(str1, str2) === 0;
+  }
+
+  gt(str1: string, str2: string): boolean {
+    return this.compare(str1, str2) > 0;
+  }
+
+  gte(str1: string, str2: string): boolean {
+    return this.compare(str1, str2) >= 0;
+  }
+
+  lt(str1: string, str2: string): boolean {
+    return this.compare(str1, str2) < 0;
+  }
+
+  lte(str1: string, str2: string): boolean {
+    return this.compare(str1, str2) <= 0;
+  }
+
+  sort(list: string[]): string[] {
```

Comment:
i could imagine it would take 1 day before we would feel the need for reverse ordering:

suggestion. Add arg reverse with default: false.

Reply:
removed sort entirely.